### PR TITLE
OCPBUGS-31685: Revert - terminal: use username if uid is not present

### PR DESCRIPTION
**Pull request by jhadvig**
Reverts https://github.com/openshift/console/commit/e87dc6fc852d23e8fca211cd0d3862b07cefa79e
/assign @stlaz

### CODE REVIEW

File: pkg/terminal/proxy.go
Comment (stlaz) [inline pkg/terminal/proxy.go:128]: this breaks oidc where UID does not have to be set and username is not going to be `kube:admin`

**Review by stlaz — CHANGES_REQUESTED** [review]

**Comment by stlaz** [comment]
/hold
we can't do this

**Comment by jhadvig** [comment]
Re-opening the PR so QE can tests if the change in the devworkspace pkg is actually breaking the external OIDC setup.

**Comment by jhadvig** [comment]
Based on the testing with [YaDan Pei](https://issues.redhat.com/secure/ViewProfile.jspa?name=rhn-support-yapei) reverting the commit is fixing only the standard OCP setup, without external OIDC. Based on chat with [Ilya Buziuk](https://issues.redhat.com/secure/ViewProfile.jspa?name=ibuziuk%40redhat.com) the webterminal is not working for the external OIDC setup with or without the PR in place.

Reverting the commit in order to unblock the webterminal team.

**Review by TheRealJon — APPROVED** [review]
/lgtm

**Pull request merged**
